File: src/hash/위장.java
package hash;

import java.util.*;

public class 위장 {

    public static int solution(String[][] clothes) {

        int answer = 1;

        HashMap<String, Integer> hm = new HashMap<>();
        Arrays.stream(clothes).forEach(c -> hm.put(c[1], hm.getOrDefault(c[1], 0)+1));

        // 안입는 경우를 포함해서 종류+1 을 전부 곱해서 경우의 수 구하기
        for (String key : hm.keySet()) {
            answer *= hm.get(key)+1;
        }

        // 모든 종류의 옷을 안입는 경우의 수 하나 빼기
        answer--;

        return answer;
    }

    public static void main(String[] args) {
        String[][] clothes = {{"yellowhat", "headgear"}, {"bluesunglasses", "eyewear"},{"green_turban", "headgear"}};
        System.out.println(solution(clothes));
    }

}

File: 0826/해시/위장.py
def solution(clothes):
    answer = 1
    wear = {}
    for c in clothes:
        # 옷 종류 당 몇개 있는지 저장
        wear[c[1]] = wear.get(c[1],0) + 1
    for w in wear:
        # 안 입을 경우 +1 해서 경우의 수 계산
        answer *= wear[w] + 1

    # 아무것도 안 입을 경우 제외    
    answer -= 1
            
    return answer
